compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -103,17 +103,18 @@ pub fn link_binary(
         });
 
         if outputs.outputs.should_link() {
-            let tmpdir = TempDirBuilder::new()
-                .prefix("rustc")
-                .tempdir()
-                .unwrap_or_else(|error| sess.dcx().emit_fatal(errors::CreateTempDir { error }));
-            let path = MaybeTempDir::new(tmpdir, sess.opts.cg.save_temps);
             let output = out_filename(
                 sess,
                 crate_type,
                 outputs,
                 codegen_results.crate_info.local_crate_name,
             );
+            let tmpdir = TempDirBuilder::new()
+                .prefix("rustc")
+                .tempdir_in(output.parent().unwrap_or_else(|| Path::new(".")))
```

Comment:
This is a similar scheme as used for rustc_metadata https://github.com/rust-lang/rust/blob/0bd13c38df9a9a922bd8ec98ba7a8bc7e111860e/compiler/rustc_metadata/src/fs.rs#L45